config/crds/azureprovider_v1alpha1_azureclusterproviderspec.yaml

```diff
@@ -474,7 +474,21 @@ spec:
               type: object
           type: object
         resourceGroup:
-          type: string
+          properties:
+            id:
+              description: ID is the identifier of the resource group.
+              type: string
+            managed:
```

Comment:
can this be a bool instead of a yes or no? ismanaged : true / ismanaged: false, default can be false.

Reply:
actually curious why isnt this implemented using tags? since we have services now, its really easy to add these tags in CreateOrUpdate?

Reply:
We need to use string values, as there's no guarantee the various tools that manipulate the yaml will interpret `"true"|"false"` as `true|false`. I use `"yes"|"no"` here instead so that there's no confusion about bool vs string.

There's a larger discussion about tagging in cloud providers vs cluster API providers (ref: https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/704, https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/706) which suggests that there still are some design decisions that need to be made, hence my hesitation to start introducing them here.

This is a lightweight solution in the interim and can refactored easily once there's some consensus about how to tag Cluster API objects.

cc-ing @johnharris85, as he intends to work on writing the tagging bits.